include/deal.II/matrix_free/operators.h

```diff
@@ -288,6 +288,48 @@ namespace MatrixFreeOperators
 
 
 
+  /**
+   * This class implements the operation of the action of a mass matrix
+   * and additionally the action of a lumped mass matrix that can be used for
+   * preconditioning. This class is used in VectorTools::project_distributed().
+   *
+   * @author Daniel Arndt, 2016
+   */
+  template <int dim, int fe_degree, int n_components = 1, typename number = double>
+  class MassOperator : public Base<dim, number>
+  {
+  public:
+
+    /**
+     * Constructor. Initializes the MatrixFree object.
+     */
+    MassOperator (const MatrixFree<dim,number> &data);
+
+    /**
+     * For preconditioning, we store a lumped mass matrix at the diagonal entries.
+     */
+    virtual void compute_diagonal ();
+
+  private:
+    /**
+     * Applies the mass matrix operation on an input array. It is
+     * assumed that the passed input and output arrays are of correct size,
```

Comment:
you don't pass, arrays but LA::distr::Vectors and the size feels wrong, it looks like the description is out of place.